CairoMakie/src/primitives.jl

```diff
@@ -287,32 +267,32 @@ function draw_marker(ctx, marker::Char, font, pos, scale, strokecolor, strokewid
     cairo_font_face_destroy(cairoface)
 
     set_font_matrix(ctx, old_matrix)
-    Cairo.restore(ctx)
-
 end
 
+function draw_marker(ctx, marker::Circle, pos, scale, strokecolor, strokewidth, marker_offset, rotation)
 
-function draw_marker(ctx, marker::Union{Rect, Type{<: Rect}}, pos, scale, strokecolor, strokewidth, marker_offset, rotation)
-    s2 = if marker isa Type{Rect}
-        Point2(scale[1], -scale[2])
-    else
-        Point2((widths(marker) .* scale .* (1, -1))...)
-    end
+    marker_offset = marker_offset + scale ./ 2
+    pos += Point2f0(marker_offset[1], -marker_offset[2])
+    Cairo.arc(ctx, pos[1], pos[2], scale[1]/2, 0, 2*pi)
+    Cairo.fill_preserve(ctx)
 
-    offset = marker_offset .+ scale ./ 2
+    Cairo.set_line_width(ctx, Float64(strokewidth))
 
-    pos += Point2f0(offset[1], -offset[2])
+    sc = to_color(strokecolor)
```

Comment:
I wanted to go more into the direction of converting these things before going into the drawing functions. It would probably be good if colors had to be RGBA already etc to avoid conversion costs in these tight loops.

Reply:
Yeah, that's what I started implementing in the MakieCore refactor ;)

Reply:
ok great then this is fine. I also made a `TorVector` like data structure in the mathtexengine pr, probably should merge that with your approach. It did make for a nice integration with `broadcast_foreach` I thought. We should probably make the attribute conversions explicit to one of these types. I find `convert_attribute(attr, key"color"())` a bit opaque, what about `convert(TorVector{RGBAf0}, attr)`? I called the data structure ScalarOrVector I think.

Reply:
the plan is to refactor all the `convert_attribute(x, key"color")` to get called by `to_color(x)` ... I already started this, but only for the most used once (e.g. `to_color, to_font` etc).

>what about convert(TorVector{RGBAf0}, attr)?

I guess when we have the new, fully typed plot types, we can do:
`field_convert(fieldtype(plot, key), x)` ... I don't think we'll be able to use `convert`, since that would need a lot of type piracy.
Also, I'm a bit afraid, that it will be harder to implement, since the unions in the field types get pretty complex, so may just stick with `to_color` etc. Also, we may still need some fields of type any...

Reply:
Btw, `convert_attribute(attr, key"color"())` is really meant as an internal interface, for operations like:
```julia
for (key, value) in attributes
   convert_attribute(value, Key{key}())
end
```

Reply:
ok but the output of these `to_color` etc. conversions could be one of these TorVectors I guess, so that the caller has to handle both possibilities correctly?

Reply:
Yeah... the main change for the new Plot types will be, that at least the set of result types should be easily visible.